bouncer/views.py

```diff
@@ -96,6 +92,54 @@ def index(request):
         location=request.registry.settings["hypothesis_url"])
 
 
+@view.view_config(renderer="bouncer:templates/annotation.html.jinja2",
+                  route_name="goto_url")
+def goto_url(request):
+    """
+    View that takes the user to a URL with the annotation layer enabled.
+
+    Optional configuration for the client may be specified via additional query
+    params:
+
+    "q" - Initial query for the filter input in the client.
+    """
+    settings = request.registry.settings
+    url = request.params.get('url')
+
+    if url is None:
+        raise httpexceptions.HTTPBadRequest('"url" parameter is missing')
+
+    # Remove any existing #fragment identifier from the URI before we
+    # append our own.
+    url = parse.urldefrag(url)[0]
```

Comment:
Personal opinion (no need to change if you disagree): I'd put this after the `_is_http_url` call, so first you have two code blocks validating the url (not empty, http(s)), and then you start messing around with it before returning the result to the browser.